### DISCUSSION TITLE
Is Formik dead?

### DISCUSSION THREAD

**Discussion by reo777** [jaredpalmer/formik#3526] [General] [answered]
There are no updates since this release https://github.com/jaredpalmer/formik/releases/tag/formik%402.2.9 .
Need more maintainers or it will be deprecated?

**Comment by tastafur** [comment]
@jaredpalmer will there be more maintenance?

**Comment by johnrom** [comment, marked as answer]
I would move on to React Hook Form if you're starting fresh. I have a fork in the works but honestly I don't have much time to work on it and it wouldn't be released for a few months (if ever)

**Reply by kelly-tock** [reply]
What happened?

**Reply by johnrom** [reply]
@kelly-tock 

- First I had to deal with the licensing issue that was the Formik switch from MIT to Apache2.
- Then I had to solve the design issues with state handling and Context optimization.
- Now I have to get type inference to 💯. Unfortunately, I've got it like 90% of the way. The extra 10% has been a few dozen hours of back and forth trying different things -- and I only work on it a few hours a week.
- Then I have to solve several design flaws with Formik, like: 
  - Reliance on child functions for state -- these should be fenced off into hooks or component wrappers that are opted into
  - The usage of Field to cover all different aspects of inputs and select boxes even though they are very different (TS can't make sense of that level of variance, and even if it could it would not be maintainable or usable from the dev side)
  - Integration of third party libraries directly in the code (they should use plugins / extensibility)
- Once all that's done, I need to make sure the API still feels as good as original Formik even with the expanded surface, and has some semblance of backwards compatibility so as not to smack the users of Formik in the face with a huge overhaul.

If I can't spin up a fork with a better dev experience than RHF, then there's not really a point to release one.

**Reply by SevenOutman** [reply]
Sounds like a reasonable point for handing-over?

**Reply by brandonlenz** [reply]
I'm working on a migration guide to React Hook Form since it seems like a necessary eventuality, and isn't _that_ difficult even for complex forms.

https://github.com/trussworks/formik-to-react-hook-form-migration-guide

I welcome feedback and improvements to it

**Comment by florianwalther-private** [comment]
Is it dangerous to keep using Formik since there hasn't been an update in over a year?

**Reply by mikepage** [reply]
With React 18 available since March 29 a lot of features are being developed in meta frameworks like React Router, Remix, Next.js. These features will improve the React DX (developer experience). You will not be able to use these features with this outdated library. 

In my experience converting an existing Formik project to React Hook Form is not too hard. You have to understand the basic principles, for example useFormContext (https://react-hook-form.com/api/useformcontext) and useController (https://react-hook-form.com/api/usecontroller).

**Reply by mikepage** [reply]
For great full app examples you should take a look at: https://github.com/alan2207/bulletproof-react/blob/master/src/components/Form/InputField.tsx

If you are starting fresh you could also try RedwoodJS, this is a full-stack framework with built-in forms and validation support:
https://redwoodjs.com/docs/forms

**Reply by florianwalther-private** [reply]
Thank you. But is it dangerous to keep using Formik? I'm not starting a new project and the migration away from Formik will require quite a lot of work for me.

**Reply by mikepage** [reply]
It is not insecure at the moment https://snyk.io/advisor/npm-package/formik

**Reply by florianwalther-private** [reply]
Thank you! I'll have to switch eventually but good to know that it's working for now. So in your opinion, React-Hook-Form is the best replacement?

**Reply by mikepage** [reply]
It is a great replacement with zero dependencies and almost no open issues
https://blog.logrocket.com/react-hook-form-vs-formik-comparison/

**Reply by florianwalther-private** [reply]
@mikepage Thank you very much!

**Comment by lekoaf** [comment]
Apparently not!

So why did it start up again?

I'm actually quite happy about this, since my team has been using Formik for a _long_ time. But since we thought it  was dead we've investigated alternatives.

Curious to know what changed.

**Reply by Ritiksh0h** [reply]
can give the alternatives links
